tests/callbacks/test_swa.py

```diff
@@ -0,0 +1,144 @@
+    swa_callback = SwaTestCallback(swa_epoch_start=swa_start, swa_lrs=0.1)
+    assert swa_callback.update_parameters_calls == 0
+    assert swa_callback.transfer_weights_calls == 0
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=max_epochs,
+        limit_train_batches=5,
+        limit_val_batches=0,
+        callbacks=[swa_callback],
+        accumulate_grad_batches=2,
+        accelerator=accelerator,
+        gpus=gpus,
+    )
+    trainer.fit(model)
+
+    # check the model is the expected
+    assert trainer.get_model() == model
+
+    # make sure these are correctly set again
+    assert not trainer.train_loop._skip_backward
+    assert trainer.accumulate_grad_batches == 2
+    assert trainer.num_training_batches == 5
+
+    # check backward call count. the batchnorm update epoch should not backward
+    assert trainer.dev_debugger.count_events("backward_call") == max_epochs * trainer.limit_train_batches
+
+    assert swa_callback.update_parameters_calls == max_epochs - (swa_start - 1)
+    assert swa_callback.transfer_weights_calls == 1
+
+
+@pytest.mark.skipif(not _PYTORCH_GREATER_EQUAL_1_6_0, reason="SWA available from in PyTorch 1.6.0")
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+@pytest.mark.skipif(not os.getenv("PL_RUNNING_SPECIAL_TESTS", '0') == '1',
+                    reason="test should be run outside of pytest")
+def test_swa_callback_ddp(tmpdir):
+    train_with_swa(tmpdir, accelerator="ddp" , gpus=2)
+
+
+@pytest.mark.skipif(not _PYTORCH_GREATER_EQUAL_1_6_0, reason="SWA available from in PyTorch 1.6.0")
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+def test_swa_callback_ddp_spawn(tmpdir):
+    train_with_swa(tmpdir, accelerator="ddp_spawn" , gpus=2)
+
+
+@pytest.mark.skipif(not _PYTORCH_GREATER_EQUAL_1_6_0, reason="SWA available from in PyTorch 1.6.0")
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="test requires a GPU machine")
+def test_swa_callback_1_gpu(tmpdir):
+    train_with_swa(tmpdir, accelerator=None, gpus=1)
+
+
+@pytest.mark.skipif(not _PYTORCH_GREATER_EQUAL_1_6_0, reason="SWA available from in PyTorch 1.6.0")
+@pytest.mark.parametrize("batchnorm", (True, False))
+def test_swa_callback(tmpdir, batchnorm):
+    train_with_swa(tmpdir, batchnorm=batchnorm)
+
+
+@pytest.mark.skipif(not _PYTORCH_GREATER_EQUAL_1_6_0, reason="SWA available from in PyTorch 1.6.0")
+def test_swa_raises():
+    with pytest.raises(MisconfigurationException):
+        StochasticWeightAveraging(swa_epoch_start=0, swa_lrs=0.1)
+    with pytest.raises(MisconfigurationException):
+        StochasticWeightAveraging(swa_epoch_start=1.5, swa_lrs=0.1)
+    with pytest.raises(MisconfigurationException):
+        StochasticWeightAveraging(swa_epoch_start=-1, swa_lrs=0.1)
+    with pytest.raises(MisconfigurationException):
+        StochasticWeightAveraging(swa_epoch_start=5, swa_lrs=[0.2, 1])
```

Comment:
you shall also match the message that correct error is raised

Reply:
`
```suggestion
    with pytest.raises(MisconfigurationException, match=">0 integer or a float between 0 and 1"):
        StochasticWeightAveraging(swa_epoch_start=0, swa_lrs=0.1)
    with pytest.raises(MisconfigurationException, match=">0 integer or a float between 0 and 1"):
        StochasticWeightAveraging(swa_epoch_start=1.5, swa_lrs=0.1)
    with pytest.raises(MisconfigurationException, match=">0 integer or a float between 0 and 1"):
        StochasticWeightAveraging(swa_epoch_start=-1, swa_lrs=0.1)
    with pytest.raises(MisconfigurationException, match="positive float or a list of positive float"):
        StochasticWeightAveraging(swa_epoch_start=5, swa_lrs=[0.2, 1])
```